File: activations/src/test/java/org/novi/activations/DateTimeActivationTest.java
package org.novi.activations;

import org.novi.core.exceptions.ConfigurationParseException;
import org.novi.core.exceptions.ContextParseException;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

class DateTimeActivationTest {

    @Test
    void evaluateForDateInBetween() throws ConfigurationParseException, ContextParseException {
        DateTimeActivation dta = new DateTimeActivation();
        boolean result = dta.whenConfiguredWith("""
                {
                    "startDateTime":"11-12-2023 12:00",
                    "endDateTime":"20-12-2023 12:00"
                }
                """).evaluateFor("""
                {
                    "org.novi.activations.DateTimeActivation.currentDateTime": "15-12-2023 12:00"
                }
                """);
        assertThat(result).isTrue();
    }

    @Test
    void evaluateForDateEqStartDate() throws ConfigurationParseException, ContextParseException {
        DateTimeActivation dta = new DateTimeActivation();
        boolean result = dta.whenConfiguredWith("""
                {
                    "startDateTime":"11-12-2023 12:00",
                    "endDateTime":"20-12-2023 12:00"
                }
                """).evaluateFor("""
                {
                    "org.novi.activations.DateTimeActivation.currentDateTime": "11-12-2023 12:00"
                }
                """);
        assertThat(result).isTrue();
    }

    @Test
    void evaluateForDateEqEndDate() throws ConfigurationParseException, ContextParseException {
        DateTimeActivation dta = new DateTimeActivation();
        boolean result = dta.whenConfiguredWith("""
                {
                    "startDateTime":"11-12-2023 12:00",
                    "endDateTime":"20-12-2023 12:00"
                }
                """).evaluateFor("""
                {
                    "org.novi.activations.DateTimeActivation.currentDateTime": "20-12-2023 12:00"
                }
                """);
        assertThat(result).isFalse();
    }

    @Test
    void evaluateForDateGtEndDate() throws ConfigurationParseException, ContextParseException {
        DateTimeActivation dta = new DateTimeActivation();
        boolean result = dta.whenConfiguredWith("""
                {
                    "startDateTime":"11-12-2023 12:00",
                    "endDateTime":"20-12-2023 12:00"
                }
                """).evaluateFor("""
                {
                    "org.novi.activations.DateTimeActivation.currentDateTime": "25-12-2023 12:00"
                }
                """);
        assertThat(result).isFalse();
    }

    @Test
    void evaluateForDateLtStartDate() throws ConfigurationParseException, ContextParseException {
        DateTimeActivation dta = new DateTimeActivation();
        boolean result = dta.whenConfiguredWith("""
                {
                    "startDateTime":"11-12-2023 12:00",
                    "endDateTime":"20-12-2023 12:00"
                }
                """).evaluateFor("""
                {
                    "org.novi.activations.DateTimeActivation.currentDateTime": "05-12-2023 12:00"
                }
                """);
        assertThat(result).isFalse();
    }
}

File: novi-activations/src/test/scala/org/novi/activations/dsl/DateTimeActivationTest.scala
package org.novi.activations.dsl

import org.assertj.core.api.Assertions.assertThat
import org.junit.Test
import org.novi.core.exceptions.{ConfigurationParseException, ContextParseException}

class DateTimeActivationTest {

  @Test
  @throws[ConfigurationParseException]
  @throws[ContextParseException]
  private[activations] def evaluateForDateInBetween(): Unit = {
    val dta: DateTimeActivation = new DateTimeActivation
    val result: Boolean = dta.apply(
      """
                {
                    "startDateTime":"11-12-2023 12:00",
                    "endDateTime":"20-12-2023 12:00"
                }
                """).evaluate(
      """
                {
                    "org.novi.activations.dsl.DateTimeActivation.currentDateTime": "15-12-2023 12:00"
                }
                """)
    assertThat(result).isTrue
  }

  @Test
  @throws[ConfigurationParseException]
  @throws[ContextParseException]
  private[activations] def evaluateForDateEqStartDate(): Unit = {
    val dta: DateTimeActivation = new DateTimeActivation
    val result: Boolean = dta.apply(
      """
                {
                    "startDateTime":"11-12-2023 12:00",
                    "endDateTime":"20-12-2023 12:00"
                }
                """).evaluate(
      """
                {
                    "org.novi.activations.dsl.DateTimeActivation.currentDateTime": "11-12-2023 12:00"
                }
                """)
    assertThat(result).isTrue
  }

  @Test
  @throws[ConfigurationParseException]
  @throws[ContextParseException]
  private[activations] def evaluateForDateEqEndDate(): Unit = {
    val dta: DateTimeActivation = new DateTimeActivation
    val result: Boolean = dta.apply(
      """
                {
                    "startDateTime":"11-12-2023 12:00",
                    "endDateTime":"20-12-2023 12:00"
                }
                """).evaluate(
      """
                {
                    "org.novi.activations.dsl.DateTimeActivation.currentDateTime": "20-12-2023 12:00"
                }
                """)
    assertThat(result).isFalse
  }

  @Test
  @throws[ConfigurationParseException]
  @throws[ContextParseException]
  private[activations] def evaluateForDateGtEndDate(): Unit = {
    val dta: DateTimeActivation = new DateTimeActivation
    val result: Boolean = dta.apply(
      """
                {
                    "startDateTime":"11-12-2023 12:00",
                    "endDateTime":"20-12-2023 12:00"
                }
                """).evaluate(
      """
                {
                    "org.novi.activations.dsl.DateTimeActivation.currentDateTime": "25-12-2023 12:00"
                }
                """)
    assertThat(result).isFalse
  }

  @Test
  @throws[ConfigurationParseException]
  @throws[ContextParseException]
  private[activations] def evaluateForDateLtStartDate(): Unit = {
    val dta: DateTimeActivation = new DateTimeActivation
    val result: Boolean = dta.apply(
      """
                {
                    "startDateTime":"11-12-2023 12:00",
                    "endDateTime":"20-12-2023 12:00"
                }
                """).evaluate(
      """
                {
                    "org.novi.activations.dsl.DateTimeActivation.currentDateTime": "05-12-2023 12:00"
                }
                """)
    assertThat(result).isFalse
  }
}
